### PR TITLE
[Bug] Fix configure_backend to account for data or timestamp links

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
It's possible for TimeSeries to link their `data` or `timestamps` fields to those of any other `TimeSeries`, thereby forming an internal link between the values to avoid having to duplicate the data

This was forgotten as a test case in the original PR for backend configuration.

- [x] TODO: add tests

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin OK should be fixed now, with test

File: src/neuroconv/tools/nwb_helpers/_configure_backend.py [hunk 1]
```diff
@@ -24,12 +24,20 @@ def configure_backend(
...
         nwbfile_object = nwbfile_objects[object_id]
+        # Table columns
         if isinstance(nwbfile_object, Data):
             nwbfile_object.set_data_io(data_io_class=data_io_class, data_io_kwargs=data_io_kwargs)
-        elif isinstance(nwbfile_object, TimeSeries):
+        # TimeSeries data or timestamps
+        elif isinstance(nwbfile_object, TimeSeries) and not isinstance(
+            nwbfile_object.fields.get(dataset_name), TimeSeries
+        ):
+            # print(f"here: \n\t{nwbfile_object.name=}\n\n\t\t{dataset_name=}")
```
Comment (h-mayorquin) [inline src/neuroconv/tools/nwb_helpers/_configure_backend.py:34]:
```suggestion
```

File: src/neuroconv/tools/nwb_helpers/_configure_backend.py [hunk 2]
```diff
@@ -24,12 +24,20 @@ def configure_backend(
         # TODO: update buffer shape in iterator, if present
 
         nwbfile_object = nwbfile_objects[object_id]
+        # Table columns
         if isinstance(nwbfile_object, Data):
             nwbfile_object.set_data_io(data_io_class=data_io_class, data_io_kwargs=data_io_kwargs)
-        elif isinstance(nwbfile_object, TimeSeries):
+        # TimeSeries data or timestamps
+        elif isinstance(nwbfile_object, TimeSeries) and not isinstance(
```
Comment (h-mayorquin) [inline src/neuroconv/tools/nwb_helpers/_configure_backend.py:31]: I would create an explicit variable for this condition.
`is_dataset_linked = isinstance(nwbfile_object.fields.get(dataset_name)`

I think that explicitly naming the condition will capture intent of what we are trying to do (self-document) and make it easier to change in the future if we need to.

**Review by h-mayorquin — COMMENTED** [review]
This looks OK. I would approve but I am weary of automerge.

I have one suggestion -feel free to ignore- and one request to remove a print statement over there.

File: src/neuroconv/tools/nwb_helpers/_configure_backend.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply src/neuroconv/tools/nwb_helpers/_configure_backend.py:34]: Haha, whoops

File: src/neuroconv/tools/nwb_helpers/_configure_backend.py (hunk 2 above)
Comment (CodyCBakerPhD) [reply src/neuroconv/tools/nwb_helpers/_configure_backend.py]: Sounds good, added that for readability

**Review by h-mayorquin — APPROVED** [review]

**Pull request merged**